engine/src/main/scala/cromwell/engine/workflow/lifecycle/MaterializeWorkflowDescriptorActor.scala

```diff
@@ -213,34 +218,36 @@ class MaterializeWorkflowDescriptorActor(serviceRegistryActor: ActorRef,
                                       sourceFiles: WorkflowSourceFilesCollection,
                                       conf: Config,
                                       workflowOptions: WorkflowOptions,
-                                      pathBuilders: List[PathBuilder]): ErrorOr[EngineWorkflowDescriptor] = {
-    val namespaceValidation: ErrorOr[ValidatedWomNamespace] = sourceFiles.workflowType match {
-      case Some(wdl) if wdl.equalsIgnoreCase("wdl") => validateWdlNamespace(sourceFiles, workflowOptions, pathBuilders)
+                                      pathBuilders: List[PathBuilder]): Parse[EngineWorkflowDescriptor] = {
+    val namespaceValidation: Parse[ValidatedWomNamespace] = sourceFiles.workflowType match {
+      case Some(wdl) if wdl.equalsIgnoreCase("wdl") => EitherT{ IO{ validateWdlNamespace(sourceFiles, workflowOptions, pathBuilders).toEither }}
       case Some(cwl) if cwl.equalsIgnoreCase("cwl") => validateCwlNamespace(sourceFiles, workflowOptions, pathBuilders)
-      case Some(other) => s"Unknown workflow type: $other".invalidNel
-      case None => "Need a workflow type here !".invalidNel
+      case Some(other) => EitherT{ IO{ s"Unknown workflow type: $other".invalidNelCheck[ValidatedWomNamespace] }}
+      case None => EitherT{ IO{ "Need a workflow type here !".invalidNelCheck[ValidatedWomNamespace] }}
     }
-    val labelsValidation = validateLabels(sourceFiles.labelsJson)
+    val labelsValidation: Parse[Labels] = EitherT{ IO{ validateLabels(sourceFiles.labelsJson).toEither }}
 
-    (namespaceValidation, labelsValidation) flatMapN { case (validatedNamespace, labels) =>
-      pushWfNameMetadataService(validatedNamespace.executable.entryPoint.name)
-      publishLabelsToMetadata(id, validatedNamespace.executable.entryPoint.name, labels)
-      buildWorkflowDescriptor(id, sourceFiles, validatedNamespace, workflowOptions, labels, conf, pathBuilders)
-    }
+    for {
+      validatedNamespace <- namespaceValidation
+      labels <- labelsValidation
+       _ <- pushWfNameMetadataService(validatedNamespace.executable.entryPoint.name)
+      _ <- publishLabelsToMetadata(id, validatedNamespace.executable.entryPoint.name, labels)
+      ewd <- EitherT{ IO{ buildWorkflowDescriptor(id, sourceFiles, validatedNamespace, workflowOptions, labels, conf, pathBuilders).toEither }}
+    } yield ewd
```

Comment:
ewd?

Reply:
as in `eitherWorkflowDescriptor` I assume. And since I dislike having the `modifierRealVarName` thing anyways I'd extra prefer to keep this over the long form :)

Reply:
`engineWorkflowDescriptor`, so not following `modifierRealVarName`, just being lazy and potentially more readable.

I go back and forth on long names, if there is consensus I will do whichever.

Reply:
I'm :+1: on acronyms for stuff like this but not everyone agrees :)